Fix CI for 3.11 kernel

donporter opened the pull request:
After rolling the private repo to 4.19 kernel, we had a little bitrot on the public repo, which is still on 3.11.10.

Fix some minor issues to get 3.11.10 support working again.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/betrfs/28)
<!-- Reviewable:end -->

donporter committed:
Fix KDIR assignment for ftfs module.

donporter committed:
Increase XFS timeout

dcowhig commented:
retest toku please

msagarpatel reviewed (changes requested):
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/betrfs/28)* status: all files reviewed, 1 unresolved discussion (waiting on @donporter)

___
*[ftfs/.mkinclude, line 3 at r1](https://reviewable.io/reviews/oscarlab/betrfs/28#-N-0to_Z1Jz62AHX7FOR:-N-0to_Z1Jz62AHX7FOS:b1yihd0) ([raw file](https://github.com/oscarlab/betrfs/blob/31446372574cf4495f50220a16309219f092b43d/ftfs/.mkinclude#L3)):*
> ```
> KDIR?=$(shell uname -r)
> #KDIR=3.11.10-ftfs
> MOD_KERN_SOURCE?=$(PWD)/../linux-3.11.10
> ```

Just mentioning: we've excised `MOD_KERN_SOURCE` from the private repo, and only use `KDIR` to point to the kernel directory. It looks like this repo still uses both, which is pretty fragile.
___
*[Jenkinsfiles/build.sh, line 7 at r1](https://reviewable.io/reviews/oscarlab/betrfs/28#-N-0r93S2qOCQwoatZII:-N-0r93S2qOCQwoatZIJ:b61hmrr) ([raw file](https://github.com/oscarlab/betrfs/blob/31446372574cf4495f50220a16309219f092b43d/Jenkinsfiles/build.sh#L7)):*
> ```Shell
> 
> vagrant up > /dev/null
> vagrant ssh -c "set -e; cd /oscar/betrfs; wget https://auku.cs.unc.edu/depot/linux-source-3.11.10-ftfs_20180618.00_all.deb; sudo dpkg -i linux-source-3.11.10-ftfs_20180618.00_all.deb; tar -xf /usr/src/linux-source-3.11.10-ftfs.tar.bz2; mv linux-source-3.11.10-ftfs linux-3.11.10; mkdir -p build; cp qemu-utils/cmake-ft.sh build/; cd build; ./cmake-ft.sh; cd ../ftfs; make KDIR=3.11.10-ftfs; rsync -vrlD /oscar/betrfs/ /vagrant/"
> ```

I was going to say that I think you may have intended to use MOD_KERN_SOURCE here, not KDIR. But then I checked ftfs/Makefile, and realized that KDIR is the right variable. OTOH, you need to define MOD_KERN_SOURCE for filesystem/Makefile...

Looking at the Jenkins log, it looks like the `sudo dpkg -i` is trying to build the kernel, which is unexpected. The kernel build appears to be failing, which prevents the pipeline from even getting to building BetrFS. Whatever the case may be, I think the right thing to do here is remove all the steps related to the kernel build since they're not relevant.


<!-- Sent from Reviewable.io -->

donporter committed:
No need to rebuild the 3.11 kernel in the benchmark job

msagarpatel committed:
drop kernel download and install

The correct kernel must already be running for this pipeline to work, so
downloading and installing the 3.11 kernel should not be necessary.

msagarpatel reviewed (commented):
> Reviewed 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/betrfs/28)* status: all files reviewed, 2 unresolved discussions (waiting on @donporter)

___
*[Jenkinsfiles/build-bench.sh, line 3 at r2](https://reviewable.io/reviews/oscarlab/betrfs/28#-N-3PcWR3VwaxteJqsmx:-N-3PcWR3VwaxteJqsmy:b-sv9q4l) ([raw file](https://github.com/oscarlab/betrfs/blob/1b0cfaae956274620e95141e895ed938235fc4a1/Jenkinsfiles/build-bench.sh#L3)):*
> ```Shell
> #!/bin/bash
> 
> vagrant ssh -c "set -e; cd /oscar/betrfs; wget https://auku.cs.unc.edu/depot/linux-source-3.11.10-ftfs_20180618.00_all.deb; sudo dpkg -i linux-source-3.11.10-ftfs_20180618.00_all.deb; mkdir -p build; cp qemu-utils/cmake-ft.sh build/; cd build; sed -i 's/DEBUG/RELEASE/g' cmake-ft.sh; ./cmake-ft.sh; cd ../filesystem; make"
> ```

Please mark this resolved if my change is okay.


<!-- Sent from Reviewable.io -->

donporter committed:
Copy the build steps for the southbound job, which works.  Build fails with missing (deprecated) pthread union header.  I think we are getting lucky in that the union file is included that dpkg/tarball.

donporter committed:
Build kernel enough to get conf headers for benchmark job

donporter committed:
Bugfix

donporter committed:
Bugfix

donporter committed:
Try building the kernel after all

donporter reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/oscarlab/betrfs/28)* status: 3 of 4 files reviewed, 2 unresolved discussions (waiting on @msagarpatel)

___
*[ftfs/.mkinclude, line 3 at r1](https://reviewable.io/reviews/oscarlab/betrfs/28#-N-0to_Z1Jz62AHX7FOR:-N-8wQhX5ldXvbzFj5qB:bguhhls) ([raw file](https://github.com/oscarlab/betrfs/blob/31446372574cf4495f50220a16309219f092b43d/ftfs/.mkinclude#L3)):*
<details><summary><i>Previously, msagarpatel (Sagar Patel) wrote…</i></summary><blockquote>

Just mentioning: we've excised `MOD_KERN_SOURCE` from the private repo, and only use `KDIR` to point to the kernel directory. It looks like this repo still uses both, which is pretty fragile.
</blockquote></details>

Yeah, let's keep it for now, but drop this when we do the next release.
___
*[Jenkinsfiles/build.sh, line 7 at r1](https://reviewable.io/reviews/oscarlab/betrfs/28#-N-0r93S2qOCQwoatZII:-N-8wVu55eWdGuhioc5V:bdmzoax) ([raw file](https://github.com/oscarlab/betrfs/blob/31446372574cf4495f50220a16309219f092b43d/Jenkinsfiles/build.sh#L7)):*
<details><summary><i>Previously, msagarpatel (Sagar Patel) wrote…</i></summary><blockquote>

I was going to say that I think you may have intended to use MOD_KERN_SOURCE here, not KDIR. But then I checked ftfs/Makefile, and realized that KDIR is the right variable. OTOH, you need to define MOD_KERN_SOURCE for filesystem/Makefile...

Looking at the Jenkins log, it looks like the `sudo dpkg -i` is trying to build the kernel, which is unexpected. The kernel build appears to be failing, which prevents the pipeline from even getting to building BetrFS. Whatever the case may be, I think the right thing to do here is remove all the steps related to the kernel build since they're not relevant.
</blockquote></details>

I had some subtle issues with removing kernel compile on the benchmark job; I'm inclined to pick up the relevant changes as part of the release.
___
*[Jenkinsfiles/build-bench.sh, line 3 at r2](https://reviewable.io/reviews/oscarlab/betrfs/28#-N-3PcWR3VwaxteJqsmx:-N-3eBEyBAn1LX3Z7XYM:b5gxkg9) ([raw file](https://github.com/oscarlab/betrfs/blob/1b0cfaae956274620e95141e895ed938235fc4a1/Jenkinsfiles/build-bench.sh#L3)):*
<details><summary><i>Previously, msagarpatel (Sagar Patel) wrote…</i></summary><blockquote>

Please mark this resolved if my change is okay.
</blockquote></details>

In this version of the code, we still need the source to build that pthread union header file.  Obviously, we should remove that as part of the release.


<!-- Sent from Reviewable.io -->

msagarpatel reviewed (approved):
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/oscarlab/betrfs/28)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @donporter)



<!-- Sent from Reviewable.io -->
